naga/src/valid/interface.rs

```diff
@@ -613,6 +615,17 @@ impl super::Validator {
         let mut argument_built_ins = crate::FastHashSet::default();
         // TODO: add span info to function arguments
         for (index, fa) in ep.function.arguments.iter().enumerate() {
+            if ep.workgroup_size[1..].iter().any(|&s| s > 1)
+                && matches!(
+                    fa.binding,
+                    Some(crate::Binding::BuiltIn(crate::BuiltIn::SubgroupId))
+                        | Some(crate::Binding::BuiltIn(
+                            crate::BuiltIn::SubgroupInvocationId
+                        ))
+                )
+            {
+                return Err(EntryPointError::InvalidMultiDimensionalSubgroupBuiltIn.with_span());
+            }
```

Comment:
We should do this in `VaryingContext.validate_impl` or else we will miss validating this in structs.

Also, do we need to disallow `SubgroupInvocationId` as well? I thought we didn't since it has equivalents in all backends.

Reply:
I looked into resolving this feedback, but I dont understand. Why would structs need validation about subgroup functions?
Also, I checked and `SubgroupInvocationId` is indeed implemented on all backends, but I think this validation is unrelated to that. Its validating the dimensionality of workgroups to make sure subgroup operations make sense, I think. What I dont understand is why subgroups arent supported on workgroup dimensionalities that are not of the form (X, 1, 1).

Reply:
That has to do with the discussion in http://github.com/gfx-rs/wgpu/pull/5301#discussion_r1534415235 , specifically the mapping between invocation in workgroup and invocation in subgroup:

> And if WorkGroupSize.x % SubgroupSize == 0 then it is also guaranteed that:
> LocalInvocationIndex = SubgroupLocalInvocationId + SubgroupId * SubgroupSize